glib/FastActivator.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace GLib
+{
+	static class FastActivator
+	{
+		const BindingFlags flags = BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.CreateInstance;
+		delegate object FastCreateObject (IntPtr ptr);
+		static FastCreateObject FastCtor (Type t)
+		{
+			FastCreateObject method;
+			if (!cache.TryGetValue (t, out method)) {
+				var param = Expression.Parameter (typeof (IntPtr));
+				var newExpr = Expression.New (
+					t.GetConstructor (flags, null, new [] { typeof (IntPtr) }, new ParameterModifier [0]),
+					param
+				);
+				cache [t] = method = (FastCreateObject)Expression.Lambda (typeof (FastCreateObject), newExpr, param).Compile ();
```

Comment:
`GLib itself is internally completely thread-safe (all global data is automatically locked),`